crates/oci-distribution/src/client.rs

```diff
@@ -266,15 +266,21 @@ impl Client {
 
         let challenge = &challenge_opt[0];
         let realm = challenge.realm.as_ref().unwrap();
-        let service = challenge.service.as_ref().unwrap();
+        let service = challenge.service.as_ref();
+        let mut query = vec![("scope", &scope)];
+
+        if service.is_some() {
```

Comment:
In the future, a more idiomatic approach would be:

```
if let Some(s) = service {
    query.push("service", s);
}
```

Using `if let` removes the extra `unwrap` call and cleans up the code to be more readable.

https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html

Reply:
Haha @bacongobbler I caught that after I merged and already did it locally. PR coming soon

Reply:
You're right @bacongobbler, I had a feeling it felt wrong but clippy wasn't saying anything. Sorry 😅